pycode/memilio-epidata/setup.py

```diff
@@ -0,0 +1,78 @@
+import sys
+import os
+import subprocess
+from setuptools import setup, find_packages, Command
+
+__version__ = '0.1.0'
+
+class PylintCommand(Command):
+    """
+    Custom command to run pylint and get a report as html.
+    """
+    description = "Runs pylint and outputs the report as html."
+    user_options = []
+
+    def initialize_options(self):
+        from pylint.reporters.text import TextReporter, ParseableTextReporter
+        from pylint.reporters.json_reporter import JSONReporter
+        from pylint_json2html import JsonExtendedReporter
+
+        self.lint_modules = ["memilio/"]
+        self.out_format = "extendedjson"
+
+        self.REPORTERS = {
+            "parseable": (ParseableTextReporter, "build_pylint/pylint_parseable.txt"),
+            "text": (TextReporter, "build_pylint/pylint.txt"),
+            "json": (JSONReporter, "build_pylint/pylint.json"),
+            "extendedjson": (JsonExtendedReporter, "build_pylint/pylint_extended.json")
+        }
+
+    def finalize_options(self):
+        self.reporter, self.out_file = self.REPORTERS.get(self.out_format)  # , self.REPORTERS.get("parseable"))
+
+    def run(self):
+        os.makedirs("build_pylint", exist_ok=True)
+
+        # Run pylint
+        from pylint import lint
+        with open(self.out_file, "w", encoding="utf-8") as report_file:
+            options = ["--rcfile=../pylintrc", *self.lint_modules]
+
+            lint.Run(options, reporter=self.reporter(report_file), do_exit=False)
+
+setup(
+    name='memilio-epidata',
+    version=__version__,
+    author='DLR-SC',
+    author_email='daniel.abele@dlr.de',
+    maintainer_email='kathrin.rack@dlr.de',
+    url='https://github.com/DLR-SC/memilio',
+    description='Part of MEmilio project, reads epidemiological data from different official and unofficial sources.',
+    entry_points={
+        'console_scripts': [
+            'getrkidata=memilio.epidata.getRKIData:main',
+            'getpopuldata=memilio.epidata.getPopulationData:main',
+            'getjhdata = memilio.epidata.getJHData:main',
+            'getdividata = memilio.epidata.getDIVIData:main',
+            'getsimdata = memilio.epidata.getSimulationData:main',
+            'cleandata = memilio.epidata.cleanData:main',
+            'getrkiestimation = memilio.epidata.getRKIDatawithEstimations:main',
+            'getcommutermigration = memilio.epidata.commuter_migration_bfa:main'
+        ],
+    },
+    packages=find_packages(where = os.path.dirname(os.path.abspath(__file__))),
+    long_description='',
+    test_suite='memilio.epidata_test',
+    install_requires= [
+        'pandas<1.2.0',
+        'matplotlib<3.4',
+        'tables',
+        'numpy<=1.19.4',
+        'openpyxl',
+        'xlrd', 
+        'requests'
+    ],
```

Comment:
```suggestion
    ],
    extras_require={"pylint": ["pylint", "pylint_json2html"],
                              "test": ["pyfakefs==4.1.0",
                                         "freezegun",
                                         "coverage"]},
```

Reply:
with this, there are two ways to install dev requirements: `pip install .[test]` or `pip install -r requirements-dev.txt`.
There should only be one way (or at least, the requirements should be only listed in one place).

I propose this:
- one extras-require `dev` which combines pylint and test. every developer needs both anyway as far as I can tell (or at least they should) and maybe sometimes installing a few unnecessary packages is better than two separate commands
- delete requirements-dev.txt, it's obsolete if requirements are all listed in setup.py